File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
  let answer = []; 
  let minArr = []

  for (let i=0; i<score.length; i++) {
    minArr.push(score[i])
    minArr.sort((a,b)=>b-a)
    if (k<minArr.length) {minArr.pop()}
    answer.push(Math.min(...minArr)) 
  }

  return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    # rank는 k길이만을 가질 수 있음
    rank, result = [], []

    for i in range(len(score)):
        rank.append(score[i])
        rank.sort()
        if i < k:
            result.append(rank[0])
        else:
            result.append(rank[-k])
            
    return result

# print(solution(3,[10, 100, 20, 150, 1, 100, 200]))
print(solution(4,[0, 300, 40, 300, 20, 70, 150, 50, 500, 1000]))
